src/dodal/beamlines/i22.py

```diff
@@ -286,107 +191,61 @@ def fswitch(
+    return HDFPanda(
+        prefix=f"{PREFIX.beamline_prefix}-EA-PANDA-01:",
         path_provider=get_path_provider(),
     )
 
 
-@skip_device()
-def panda2(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> HDFPanda:
-    return device_instantiation(
-        HDFPanda,
-        "panda2",
-        "-EA-PANDA-02:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@device_factory(skip=True)
+def panda2() -> HDFPanda:
+    return HDFPanda(
+        prefix=f"{PREFIX.beamline_prefix}-EA-PANDA-02:",
         path_provider=get_path_provider(),
     )
 
 
-@skip_device()
-def panda3(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> HDFPanda:
-    return device_instantiation(
-        HDFPanda,
-        "panda3",
-        "-EA-PANDA-03:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@device_factory(skip=True)
+def panda3() -> HDFPanda:
+    return HDFPanda(
+        prefix=f"{PREFIX.beamline_prefix}-EA-PANDA-03:",
         path_provider=get_path_provider(),
     )
 
 
-@skip_device()
-def panda4(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> HDFPanda:
-    return device_instantiation(
-        HDFPanda,
-        "panda4",
-        "-EA-PANDA-04:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@device_factory(skip=True)
+def panda4() -> HDFPanda:
+    return HDFPanda(
+        prefix=f"{PREFIX.beamline_prefix}-EA-PANDA-04:",
         path_provider=get_path_provider(),
     )
 
 
-def oav(
-    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
-) -> AravisDetector:
-    return device_instantiation(
-        NXSasOAV,
-        "oav",
-        "-DI-OAV-01:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@device_factory()
+def oav() -> AravisDetector:
+    metadata_holder = NXSasMetadataHolder(
+        x_pixel_size=(3.45e-3, "mm"),  # Double check this figure
+        y_pixel_size=(3.45e-3, "mm"),
+        description="AVT Mako G-507B",
+        distance=(-1.0, "m"),
+    )
+    return NXSasOAV(
+        prefix=f"{PREFIX.beamline_prefix}-DI-OAV-01:",
         drv_suffix="DET:",
-        hdf_suffix="HDF5:",
-        metadata_holder=NXSasMetadataHolder(
-            x_pixel_size=(3.45e-3, "mm"),  # Double check this figure
-            y_pixel_size=(3.45e-3, "mm"),
-            description="AVT Mako G-507B",
-            distance=(-1.0, "m"),
-        ),
+        hdf_suffix=HDF5_PREFIX,
         path_provider=get_path_provider(),
+        metadata_holder=metadata_holder,
     )
 
 
-@skip_device()
-def linkam(
-    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
-) -> Linkam3:
-    return device_instantiation(
-        Linkam3,
-        "linkam",
-        "-EA-TEMPC-05",
-        wait_for_connection,
-        fake_with_ophyd_sim,
-    )
+@device_factory()
+def linkam() -> Linkam3:
+    return Linkam3(prefix=f"{PREFIX.beamline_prefix}-EA-TEMPC-05")
```

Comment:
Could: I suspect we need a `:` after the prefix here. Hard to tell 100% as it doesn't connect either way and was previously being skipped so we would have missed it anyway

```suggestion
    return Linkam3(prefix=f"{PREFIX.beamline_prefix}-EA-TEMPC-05:")
```